### ISSUE TITLE
Add option to decide output directory for CMake.

### ISSUE THREAD

**Issue by gsa-tspaere** [prefix-dev/pixi-build-backends#39]
Hey!

I am using `pixi-build`  with a cmake project, and I'm able to build it and everything, but I have a problem. I use `compile_commands.json` to use `clangd` in my editor, and I'm dependent on the output of the cmake build is in a predictable place, to be order to point it there.

For now I see no option to direct pixi, to do the building in a given directory. Is there anyway you could add support for this or some other special option to get the `compile_commands.json` out?

**Comment by tdejager** [comment]
Yes! We can add the `-DCMAKE_EXPORT_COMPILE_COMMANDS=ON` option by default I think. Thank for giving this a test drive 🚗 🚤 !

**Comment by gsa-tspaere** [comment]
Adding it by default is good, but the main problem is that we need some way to place it in a fixed place, so the user can point their `clangd` configuration to it :)

**Comment by tdejager** [comment]
Yeah, indeed, so I don't know how you are using it, but we do re-use the build directory in the the `.pixi` folder :) the problem is that can dissappear whenever the backend decides but we could give that a first try maybe?

**Comment by gsa-tspaere** [comment]
Hey, that's very nice! Could you give me the name of the folder that is used for the build?

**Comment by tdejager** [comment]
It should be in `.pixi/build-v0/` and then the name of your project and some string. E.g for its this for some project:

```
.pixi/build-v0/cpp-sdl-rxtTGpY4RzI/bld/rattler-build_sdl_example/build
```

This contains the `compile_commands.json`.

**Comment by tdejager** [comment]
Closing this because this should work.
